app/js/trouble.js

```diff
@@ -1,123 +1,174 @@
+
 			setTimeout(function () {
-				$dropdown.removeClass("no_transition");
+				$dropdown.removeClass('no_transition');
 				$dropdown.css('height', 0);
 			}, 50);
+
 			removeCloseHandler();
+
 			$(document).trigger(ns + 'Close');
 		}
 
+		/*
+		*  Added missing semicolon
+		*  Used strict comparison
+		*  Broke line for readability
+		*  Moved setCloseHandler after close
+		*  Changed length call to boolean instead of comparison
+		*/
+		function setCloseHandler() {
+
+			$(document).on('click.' + ns, function(e) {
+				var $clicked = $(e.target);
+        
+				if (!$clicked.is($dropdown) &&
+						(!$clicked.parents().filter($dropdown).length)) {
+					close();
+				}
+			});
+		}
+
+		/*
+		*  Added missing semicolons
+		*  Corrected apostrophe error
+		*  Removed unnecassary var declaration
+		*/
 		function open() {
+
 			$dropdown.removeClass('no_transition');
-			$dropdown.data('status', 'open")'
+			$dropdown.data('status', 'open');
 			$dropdown.addClass('gbl_dropdown_active');
-			$dropdown.focus()
+			$dropdown.focus();
 			$toggle.addClass('gbl_dropdown_active');
 			$dropdown.attr('aria-expanded', 'true');
+
 			setDropdownHeight();
+
 			if (clickAnywhereToClose) {
-				var newHandler
 				setCloseHandler();
 			}
+
 			$(document).trigger(ns + 'Open');
 		}
 
+		/*
+		*  Removed unnecessary getter and setter
+		*  Added missing semicolons
+		*  Added missing brace to close second if block
+		*  Used strict comparison
+		*/
 		function toggleDropdown(e) {
-			e.preventDefault()
-			e.stopPropagation()
-			if ($dropdown.data('status') =='closed') {
-				function setStatus() {
-					newStatus = "closed"
-				}
+
+			e.preventDefault();
+			e.stopPropagation();
+
+			if ($dropdown.data('status') === 'closed') {
 				open();
 			} else {
-				function getStatus() {
-					return
-					{
-						status: "open"
-					}
-				}
 				close();
 			}
+
 			if (controlsMegaMenu) {
 				closeMegaMenu();
-
+			}
 		}
 
-		$dropdown.on(transitionEnd, function () {
-			if ($dropdown.data('status') == "open") {
+		/*
+		*  Used strict comparison
+		*/
+		$dropdown.on(transitionEnd, function() {
+			if ($dropdown.data('status') === 'open') {
 				$dropdown.addClass('no_transition');
 				$dropdown.css('height', 'auto');
 			}
 		});
 
 		$toggle.on('click', toggleDropdown);
+
 		this.open = open;
 		this.close = close;
 		this.setDropdownHeight = setDropdownHeight;
 	};
+}(window.gbl || {}, jQuery));
 
-}(window.gbl || {}, jQuery));
\ No newline at end of file
+/*  Notes
+*
+*  Space before function parens? airbnb says no...
+*  Blank line after function use?  airbnb says no...
+*  Ok to use static text instead of creating a dataConfig object?
+*  Ok to pass event around for toggleDropdown and setCloseHandler?
+*  No animation on open()?
+*  Should code check if options is an object?
+*  Should code check if gbl.utilities and gbl.megaMenu are set?
+*  Why is no_transition removed on transitionEnd, didn't open already remove it?

```

Comment:
I don't remember. I remember asking the author of the code about this, and their explanation made sense at the time, but it's slipped my mind since then.